### PR TITLE
[Flink] Add deletion-vector read/write/merge support

### PR DESCRIPTION
**Pull request by harperjiang**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [x] Flink
- [ ] Kernel
- [ ] Other

## Description

First step toward deletion-vector (DV) support in the Flink connector. Adds the read/write/merge primitives under `io.delta.flink.kernel.dv`:

- `DVAccess` (abstract): two-method contract (`write(Engine, path, RoaringBitmapArray)`, `read(Engine, path) -> RoaringBitmapArray`) plus a shared CRC32 helper and a reflective `FileIO` accessor (Kernel's public `Engine` interface does not expose `FileIO`; we hand-roll a one-shot `DefaultEngine.fileIO` field lookup, cached once at class load).
- `BinDVAccess`: implements the contract for Delta's plain `.bin` framing (`[version][length BE][payload][crc32 BE]`). Ported from `org.apache.spark.sql.delta.storage.dv.DeletionVectorStore`.
- `PuffinDVAccess`: stub for the Puffin-wrapped DV format; throws today, slated for a follow-up.
- `RoaringBitmapArray`: minimal Java port of `org.apache.spark.sql.delta.deletionvectors.RoaringBitmapArray`, trimmed to the surface this connector actually needs (`add`/`addAll`, `contains`, `cardinality`, `isEmpty`, `merge`, `toArray`, `serializeAsByteArray`, static `readFrom`, plus `equals`/`hashCode`). Only the Portable serialization format is supported -- delta-spark writes Portable in all 8 production+test call sites we checked, so the Native format would be dead code here.

Part of #5901.

## How was this patch tested?

- `BinDVAccessTest` -- round-trip + edge cases (empty bitmap, dense/sparse keys, `putIfAbsent` rejection, missing file, corrupted checksum, bad version byte, Portable magic byte assertion).
- `RoaringBitmapArrayTest` -- add/contains, sorted `toArray`, merge as union, equality, serialization round-trip, magic-number constant lock.
- `build/sbt \"flink/Test/compile\"` and `build/sbt \"flink/Compile/doc\"` to be validated on CI.

## Does this PR introduce _any_ user-facing changes?

No new public API surface for the connector yet -- everything lands in the `io.delta.flink.kernel.dv` package. Future PRs will wire these primitives into the writer's merge path.

### CODE REVIEW

File: flink/src/main/java/io/delta/flink/kernel/dv/BinDVAccess.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  /**
+   * Reads the single deletion vector from {@code filePath}, starting at the beginning of the file.
+   *
+   * <p>Only single-DV files are supported: the file must consist of exactly {@code
+   * [version][length BE][payload][crc32 BE]} and nothing more. If the file is longer than one DV
+   * (e.g. a delta-spark bin-packed file holding multiple DVs), this throws rather than returning a
+   * truncated or wrong result. See the class Javadoc for the multi-DV limitation.
+   */
+  @Override
+  public RoaringBitmapArray read(Engine engine, String filePath) {
```
Comment (KaiqiJinWow) [inline flink/src/main/java/io/delta/flink/kernel/dv/BinDVAccess.java:100]: Can we make the read side descriptor/range-aware here? The current `read(engine, filePath)` API ignores `DeletionVectorDescriptor.offset` and `sizeInBytes`, then enforces that the whole `.bin` file contains exactly one DV. That means this cannot read Spark-written bin-packed DV files: even a descriptor pointing to the first DV will fail when the file contains a second DV, and descriptors with non-zero offsets cannot be read at all.

This also weakens the interop claim in the PR description and will affect the follow-up MoR upsert path when it tries to merge with an existing DV written by Spark.

Could we expose something like `read(engine, DeletionVectorDescriptor, tableRoot)` or `read(engine, path, offset, size, cardinality)`, seek to the descriptor offset, read exactly `[length][payload][crc]`, verify `length == sizeInBytes` and `bitmap.cardinality() == descriptor.cardinality`, and not require EOF? A test with one `.bin` file containing two DVs would catch this.

File: flink/src/test/java/io/delta/flink/kernel/dv/BinDVAccessTest.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+/**
+ * Round-trip + edge-case tests for {@link BinDVAccess}.
+ *
+ * <p>{@link BinDVAccess} no longer carries its own Hadoop {@code Configuration}; the {@link Engine}
+ * we pass in wires up the Hadoop-backed {@code FileIO} for both write and read paths.
+ */
+class BinDVAccessTest extends TestHelper {
+
+  @Test
```
Comment (KaiqiJinWow) [inline flink/src/test/java/io/delta/flink/kernel/dv/BinDVAccessTest.java:43]: Could we add a descriptor-level test around `store()` as well? The current coverage mostly round-trips through `write()` / `read(path)`, but the important contract for the follow-up upsert path is the descriptor returned by `store()`.

It would be useful to assert that the descriptor is path-based (`storageType = "p"`), carries an absolute `.bin` path, has `offset = 1`, has `sizeInBytes` equal to the serialized payload length, and has `cardinality` matching the bitmap. Once the read API becomes descriptor/range-aware, the same test can read back through the descriptor path too.

**Comment by KaiqiJinWow** [comment]
Small PR-description cleanup: the body still mentions a `PuffinDVAccess` stub and a `putIfAbsent` rejection test, but the current diff does not include `PuffinDVAccess` and I do not see that test in `BinDVAccessTest`. Could you refresh the description so reviewers do not go looking for files/tests that are no longer part of this PR?

File: flink/src/main/java/io/delta/flink/kernel/dv/BinDVAccess.java (hunk 1 above)
Comment (harperjiang) [reply flink/src/main/java/io/delta/flink/kernel/dv/BinDVAccess.java:100]: We explicitly choose to only support single DV in the file for now as that is the primary use case here.

There are tests and clear error messages saying this.

Supporting for multi-DV in one file will come later.

File: flink/src/test/java/io/delta/flink/kernel/dv/BinDVAccessTest.java (hunk 2 above)
Comment (harperjiang) [reply flink/src/test/java/io/delta/flink/kernel/dv/BinDVAccessTest.java:43]: Added

**Review by KaiqiJinWow — APPROVED** [review]

**Review by murali-db — APPROVED** [review]

**Comment by harperjiang** [comment]
> Small PR-description cleanup: the body still mentions a `PuffinDVAccess` stub and a `putIfAbsent` rejection test, but the current diff does not include `PuffinDVAccess` and I do not see that test in `BinDVAccessTest`. Could you refresh the description so reviewers do not go looking for files/tests that are no longer part of this PR?

Done. Thanks for the suggestion!
